src/Microsoft.DotNet.XHarness.CLI/Commands/WebServer.cs

```diff
@@ -3,31 +3,29 @@
 // See the LICENSE file in the project root for more information.
 
 using System;
-using System.Net.WebSockets;
+using System.Collections.Generic;
 using System.Linq;
+using System.Net.WebSockets;
+using System.Runtime.Loader;
 using System.Threading;
 using System.Threading.Tasks;
-
+using Microsoft.AspNetCore.Builder;
 using Microsoft.AspNetCore.Hosting;
 using Microsoft.AspNetCore.Hosting.Server.Features;
-using Microsoft.Extensions.Logging;
-using Microsoft.Extensions.DependencyInjection;
-
-using LogLevel = Microsoft.Extensions.Logging.LogLevel;
-using System.Runtime.Loader;
-using System.Collections.Generic;
-using Microsoft.AspNetCore.Builder;
 using Microsoft.AspNetCore.Routing;
 using Microsoft.AspNetCore.StaticFiles;
+using Microsoft.DotNet.XHarness.CLI.CommandArguments.Wasm;
+using Microsoft.Extensions.DependencyInjection;
 using Microsoft.Extensions.FileProviders;
+using Microsoft.Extensions.Logging;
 using Microsoft.Extensions.Options;
-using Microsoft.DotNet.XHarness.CLI.CommandArguments;
+using LogLevel = Microsoft.Extensions.Logging.LogLevel;
 
 namespace Microsoft.DotNet.XHarness.CLI.Commands
 {
     public class WebServer
     {
-        internal static async Task<ServerURLs> Start(TestCommandArguments arguments, ILogger logger, Func<WebSocket, Task>? onConsoleConnected, CancellationToken token)
+        internal static async Task<ServerURLs> Start(IWebServerArguments arguments, ILogger logger, Func<WebSocket, Task>? onConsoleConnected, CancellationToken token)
```

Comment:
@pavelsavara this refactoring uncovered some problems in the code that were not visible before. One of them is that the `wasm test` command doesn't require the `--app` argument but the `wasm test-browser` does. Example logs from our CI:
```
XHarness command issued: wasm test --engine=V8 --js-file=runtime.js -v --output-directory=/datadisks/disk1/work/C00109F8/w/BB980A05/uploads -- --enable-gc --run WasmTestRunner.dll System.Reflection.Extensions.Tests.dll -notrait category=IgnoreForCI -notrait category=OuterLoop -notrait category=failing
[11:57:57] fail: Required argument app|a= was not supplied
```
https://helixre8s23ayyeko0k025g8.blob.core.windows.net/dotnet-xharness-refs-pull-630-merge-da7e4aeb8cb441b7ad/System.Reflection.Extensions.Tests/console.5ea40d4b.log?sv=2019-07-07&se=2021-06-29T11%3A55%3A57Z&sr=c&sp=rl&sig=kekuKVADunM6uhxCl8YXovs0qeY9Jc4i3t2Nfxj0hmk%3D

So, before, if we tried to spawn a web server with `wasm test` here, it would blow up on line 36:
```csharp
            var host = new WebHostBuilder()
                .UseKestrel()
                .UseContentRoot(arguments.AppPackagePath)     <<<<<<<<<<<<<<<<<<<<<<<
                .UseStartup<TestWebServerStartup>()
```

Now we have more strict validation and it uncovered this.

Can you please show me the usage of these commands in the runtime repo so that I can check we're really not using `--app` in one of them?

How should we handle this?

Reply:
I can see multiple options:
a) make `.UseContentRoot` only part of the `wasm test-browser` path. 
b) roll back the change which added the server to `wasm test`. Downside is that 'normal' scenario tests in inner loop could rely on this.
c) fid the proper directory which need to be passed to wasm test via `--app` and do it too.

In any case we should make sure that the script doesn't die silently. I'm confused how I missed it.

Reply:
I will do `a)` as part of this PR

Reply:
https://github.com/dotnet/xharness/pull/630/commits/94bdf98cb712cf6a3dbc4947959a5430701a9df2

Reply:
`--app` is set [here](https://github.com/dotnet/runtime/blob/main/eng/testing/tests.wasm.targets#L21-L22). It isn't useful for `wasm test`. IIRC, it was kept around because it was difficult to separate the changes from `wasm test-browser` in `runtime`.

We probably need do some clean up on the runtime side to be able to stop passing the arg. So, (a) makes sense.

Reply:
Thanks for the input! I can bump the XHarness in a separate PR and watch the build